servers/rendering/renderer_scene_cull.cpp

```diff
@@ -2119,58 +2119,83 @@ bool RendererSceneCull::_light_instance_update_shadow(Instance *p_instance, cons
 	return animated_material_found;
 }
 
-void RendererSceneCull::render_camera(RID p_render_buffers, RID p_camera, RID p_scenario, Size2 p_viewport_size, float p_screen_lod_threshold, RID p_shadow_atlas) {
-// render to mono camera
+void RendererSceneCull::render_camera(RID p_render_buffers, RID p_camera, RID p_scenario, Size2 p_viewport_size, float p_screen_lod_threshold, RID p_shadow_atlas, Ref<XRInterface> &p_interface) {
 #ifndef _3D_DISABLED
+	// render to mono camera
 
 	Camera *camera = camera_owner.getornull(p_camera);
 	ERR_FAIL_COND(!camera);
 
-	/* STEP 1 - SETUP CAMERA */
-	CameraMatrix camera_matrix;
-	bool ortho = false;
-
-	switch (camera->type) {
-		case Camera::ORTHOGONAL: {
-			camera_matrix.set_orthogonal(
-					camera->size,
-					p_viewport_size.width / (float)p_viewport_size.height,
-					camera->znear,
-					camera->zfar,
-					camera->vaspect);
-			ortho = true;
-		} break;
-		case Camera::PERSPECTIVE: {
-			camera_matrix.set_perspective(
-					camera->fov,
-					p_viewport_size.width / (float)p_viewport_size.height,
-					camera->znear,
-					camera->zfar,
-					camera->vaspect);
-			ortho = false;
+	SceneCamera cameras[2];
+	uint32_t num_cameras = 1;
 
-		} break;
-		case Camera::FRUSTUM: {
-			camera_matrix.set_frustum(
-					camera->size,
-					p_viewport_size.width / (float)p_viewport_size.height,
-					camera->offset,
-					camera->znear,
-					camera->zfar,
-					camera->vaspect);
-			ortho = false;
-		} break;
+	if (p_interface.is_valid()) {
+		bool is_stereo = p_interface->is_stereo();
+
+		float aspect = p_viewport_size.width / (float)p_viewport_size.height;
+		cameras[0].projection = p_interface->get_projection_for_eye(is_stereo ? XRInterface::EYE_LEFT : XRInterface::EYE_MONO, aspect, camera->znear, camera->zfar);
```

Comment:
I wonder if we should follow the example of the mulitview `ViewIndex` variable and have `EYE_MONO` and `EYE_LEFT` be the same value.
For one it would simplify the logic below, as you could use the matching defined constant as an index in the `SceneCamera` array.

Reply:
The problem is that `EYE_MONO` and `EYE_LEFT` are not the same. `EYE_MONO` provides a center location of the head, `EYE_LEFT` is centered on the camera/eye.

So while we will have ViewIndex be 0 for Mono/Left and 1 for Right, it doesn't align with these variables. Bit of a pain I don't really have an answer for right now.

Reply:
I've changed this seeing I'm looking at putting the ground work in for having more then 2 views. 

`is_stereo` is being replaced by `get_view_count`, we'll have a `get_camera_transform` to get the position and orientation of the HMD/Phone/etc and then we have `get_projection_for_view` and `get_transform_for_view` replacing resp. `get_projection_for_eye` and `get_transform_for_eye`. Both new functions take a simple `p_view` input.